src/coreclr/src/jit/emitarm64.cpp

```diff
@@ -14882,4 +14873,81 @@ emitter::insExecutionCharacteristics emitter::getInsExecutionCharacteristics(ins
 
 #endif // defined(DEBUG) || defined(LATE_DISASM)
 
+//----------------------------------------------------------------------------------------
+// IsRedundantMov:
+//    Check if the current `mov` instruction is redundant and can be omitted.
+//    A `mov` is redundant in following 3 cases:
+//
+//    1. Move to same register
```

Comment:
The comment here could also point out that `mov w1,w1` can't be removed because it zeros the upper bits of the x1 register. And similar for SIMD registers.